crates/goose/src/providers/githubcopilot.rs

```diff
@@ -132,6 +132,31 @@ pub struct GithubCopilotProvider {
 }
 
 impl GithubCopilotProvider {
+    fn payload_contains_image(payload: &Value) -> bool {
+        if let Some(messages) = payload.get("messages").and_then(|m| m.as_array()) {
+            for msg in messages {
+                if let Some(content) = msg.get("content") {
```

Comment:
you could do this more functional, like messages.iter.any and then cast the is_object to a single list but up to you